src/sync/arc.rs

```diff
@@ -25,8 +25,33 @@ impl<T> Arc<T> {
     }
 
     /// Returns the inner value, if the `Arc` has exactly one strong reference.
-    pub fn try_unwrap(_this: Arc<T>) -> Result<T, Arc<T>> {
-        unimplemented!();
+    #[track_caller]
+    pub fn try_unwrap(this: Arc<T>) -> Result<T, Arc<T>> {
+        if !this.obj.get_mut(location!()) {
+            return Err(this);
+        }
+
+        assert_eq!(1, std::sync::Arc::strong_count(&this.value));
+        // work around our inability to destruct the object normally,
+        // because of the `Drop` presense.
+        this.obj.ref_dec(location!());
+        this.unregister();
+
+        // Use the same patter of unwrapping as `std` does.
```

Comment:
typo:
```suggestion
        // Use the same pattern of unwrapping as `std` does.
```

Reply:
also, it could be worth adding a github link to the similar code in std? not a blocker though.